### PR TITLE
Update for GHC 7.10

### PR DESCRIPTION
**Pull request by alanz**
Introduce missing Functor and Applicative instances.

By the way, Control.Monad.Maybe seems to be a clone of the one in transformers. Is this needed?

### CODE REVIEW

File: Control/Monad/Maybe.hs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
```
Comment (jkoppel) [inline Control/Monad/Maybe.hs]: What's this for?

File: Data/Generics/Strafunski/StrategyLib/MonadicFunctions.hs [hunk 2]
```diff
@@ -25,6 +25,13 @@ newtype Id a = Id a
 unId 		:: Id a -> a
 unId (Id x) 	 = x
 
+instance Functor Id where
```
Comment (jkoppel) [inline Data/Generics/Strafunski/StrategyLib/MonadicFunctions.hs]: Doesn't this already exist? Else how would the monad instance exist?

File: Control/Monad/Maybe.hs (hunk 1 above)
Comment (alanz) [inline Control/Monad/Maybe.hs]: Oops, unneccesary, thought I needed it for an import of GHC.Base.Alternative

File: Data/Generics/Strafunski/StrategyLib/MonadicFunctions.hs (hunk 2 above)
Comment (alanz) [inline Data/Generics/Strafunski/StrategyLib/MonadicFunctions.hs]: I only added it if the compiler complained. You can try remove it if you like, but pretty sure it is needed

**Comment by jkoppel** [comment]
Addressed in https://github.com/jkoppel/Strafunski-StrategyLib/commit/6b456fca9a05ad23c7f5557740b491b2e004d9f7 . New version uploaded to Hackage.

**Comment by alanz** [comment]
I don't see the new version on hackage?

http://hackage.haskell.org/package/Strafunski-StrategyLib

**Comment by jkoppel** [comment]
Strange; I definitely saw it there yesterday

**Comment by jkoppel** [comment]
Uploaded again: http://hackage.haskell.org/package/Strafunski-StrategyLib-5.0.0.5

**Comment by jkoppel** [comment]
Uploaded again;
http://hackage.haskell.org/package/Strafunski-StrategyLib-5.0.0.5

**Comment by alanz** [comment]
I see it now, thanks

**Pull request closed without merging**